site/en/docs/extensions/mv3/manifest/content_scripts/index.md

```diff
@@ -0,0 +1,262 @@
+```
+{% endCompare %}
+
+{% Compare "worse", "Does not match" %}
+```text
+https://my.example.com/foo/bar
+https://example.com/foo/
+https://www.example.com/foo
+```
+{% endCompare %}
+</div>
+
+{% Label %}manifest.json{% endLabel %}
+
+```json/5
+{
+  ...
+  "content_scripts": [
+    {
+      "matches": ["https://*.nytimes.com/*"],
+      "include_globs": ["*nytimes.com/???s/*"],
+      "js": ["content-script.js"]
+    }
+  ],
+  ...
+}
+```
+
+<div class="switcher">
+{% Compare 'better', 'Matches' %}
+```text
+https://www.nytimes.com/arts/index.html
+https://www.nytimes.com/jobs/index.html
+```
+{% endCompare %}
+
+{% Compare "worse", "Does not match" %}
+```text
+https://www.nytimes.com/sports/index.html
+https://www.nytimes.com/music/index.html
+```
+{% endCompare %}
+</div>
+
+#### `"exclude_globs"` {: #excl-globs }
+
+{% Label %}manifest.json{% endLabel %}
+
+```json/5
+{
+  ...
+  "content_scripts": [
+    {
+      "matches": ["https://*.nytimes.com/*"],
+      "exclude_globs": ["*science*"],
+      "js": ["content-script.js"]
+    }
+  ],
+  ...
+}
+```
+
+<div class="switcher">
+{% Compare 'better', 'Matches' %}
+```text
+https://history.nytimes.com
+https://.nytimes.com/music
+```
+{% endCompare %}
+
+{% Compare "worse", "Does not match" %}
+```text
+https://science.nytimes.com
+https://www.nytimes.com/science
+```
+{% endCompare %}
+</div>
+
+#### Advanced customization example {: #all-custom }
+
+{% Label %}manifest.json{% endLabel %}
+
+```json/5-7
+{
+  ...
+  "content_scripts": [
+    {
+      "matches": ["https://*.nytimes.com/*"],
+      "exclude_matches": ["*://*/*business*"],
+      "include_globs": ["*nytimes.com/???s/*"],
+      "exclude_globs": ["*science*"],
+      "js": ["content-script.js"]
+    }
+  ],
+  ...
+}
+```
+
+<div class="switcher">
+{% Compare 'better', 'Matches' %}
+```text
+https://www.nytimes.com/arts/index.html
+https://.nytimes.com/jobs/index.html
+```
+{% endCompare %}
+
+{% Compare "worse", "Does not match" %}
+```text
+https://science.nytimes.com
+https://www.nytimes.com/jobs/business
+https://www.nytimes.com/science
+```
+{% endCompare %}
+</div>
+
+{% endDetails %}
+
+## Frames {: #frames }
+
+The `"all_frames"` field specifies if the content script should be injected into all frames matching the specified URL requirements. If set to `false` it will only inject into the topmost frame. It can be used along with `"match_about_blank"` to inject into an `about:blank` frame. 
+
+To inject into other frames like `data:`, `blob:`, and `filesystem:`, set the `"match_origin_as_fallback"` to `true`. For details, see [Injecting in related frames][cs-inject-frames]
+
+`"all_frames"` Boolean
+: _Optional_. Defaults to `false`, meaning that only the top frame is matched.If specified true, it will inject into all frames, even if the frame is not the topmost frame in the tab. Each frame is checked independently for URL requirements, it won't inject into child frames if the URL requirements are not met.
```

Comment:
```suggestion
: _Optional_. Defaults to `false`, meaning that only the top frame is matched. If set to true, it will inject into all frames, even if the frame is not the topmost frame in the tab. Each frame is checked independently for URL requirements, it won't inject into child frames if the URL requirements are not met.
```